data/src/client.rs

```diff
@@ -87,6 +87,7 @@ pub struct Client {
     supports_labels: bool,
     supports_away_notify: bool,
     highlight_blackout: HighlightBlackout,
+    isupport_parameters: HashMap<String, isupport::Parameter>,
```

Comment:
After removing `Negation`, we should just make this a `HashSet<Parameter>` so it's always strongly typed.

Reply:
OR, we have `isupport::Kind` enum which lists just the param names which we care to track in Halloy and then `HashMap` that. `HashSet<Parameter>` is actually dangerous because it'd allow storing >1 parameter kind if the values are different inside the parameter.

Reply:
I see the value in `HashMap<Kind,Parameter>`, though it's a shame we lose some of the simplicity that `Remove(String)` had.  I'll probably add a function to `Operation` that returns a `Kind`, but will play around to see if there's an even cleaner way first.